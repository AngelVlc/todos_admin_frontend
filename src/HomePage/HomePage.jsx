import React from 'react';
import { useAlert } from 'react-alert'

function HomePage() {
    const alert = useAlert()

    return (
        <div>
            <h4>HOME PAGE</h4>

            <p>Backend url: {process.env.REACT_APP_BACKEND_URL}</p>

            <button
            onClick={() => {
              alert.show('Oh look, an alert!')
            }}
          >
            Show Alert
          </button>
        </div>
    );
}

export { HomePage };