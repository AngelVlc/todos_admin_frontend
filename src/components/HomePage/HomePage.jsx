import React, { useEffect, useContext, useState } from 'react';
import { Link, useHistory } from 'react-router-dom';
import { AppContext } from '../../contexts/AppContext';
import { doGet } from '../../helpers/api';

export const HomePage = () => {
  const { auth, requestsDispatch } = useContext(AppContext)
  const [users, setUsers] = useState([]);
  let history = useHistory();

  useEffect(() => {
    const getUsers = async () => {
      const res = await doGet('users', auth.info.token, requestsDispatch)
      setUsers(res);
    }
    if (auth.info) {
      getUsers();
    }
  }, [auth.info, requestsDispatch]);

  return (
    <div className="container">
      <h3 className="title">USERS</h3>
      <table className="table">
        <thead>
          <tr>
            <td>User</td>
            <td>Admin?</td>
            <td>
              <button className="button is-small" data-testid="addNew" onClick={() => history.push('user/new')}>
                <span className="icon is-small">
                  <i className="fas fa-plus"></i>
                </span>
              </button>
            </td>
          </tr>
        </thead>
        <tbody>
          {users.length > 0 && users.map((user) => (
            <tr key={user.id}>
              <td>
                <Link className="has-text-black" data-testid={`editUser${user.id}`} to={`/user/${user.id}/edit`}>{user.name}</Link>
              </td>
              <td>
                {user.isAdmin &&
                  <center>
                    <span className="icon is-small">
                      <i className="fas fa-check fa-xs"></i>
                    </span>
                  </center>
                }
              </td>
              <td>
                <center>
                  <Link className="has-text-black" data-testid={`deleteUser${user.id}`} to={`/user/${user.id}/delete`}>
                    <span className="icon is-small">
                      <i className="fas fa-trash-alt fa-xs"></i>
                    </span>
                  </Link>
                </center>
              </td>
            </tr>
          ))
          }
        </tbody>
      </table>
    </div>
  );
}
