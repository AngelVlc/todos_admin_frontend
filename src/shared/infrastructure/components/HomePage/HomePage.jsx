import React, { useContext } from 'react';
import { useHistory } from "react-router-dom";
import { AppContext } from '../../contexts/AppContext';

export const HomePage = () => {
    const { auth } = useContext(AppContext)
    let history = useHistory();

    return (
        <div className="container">
            <h3 className="title">HOME</h3>
            {auth.info?.isAdmin &&
                <div className="box">
                    <h6 className="subtitle is-6">Admin</h6>
                    <div className="buttons">
                        <button className="button is-small" onClick={() => history.push('/users')} data-testid="users">Users</button>
                        <button className="button is-small" onClick={() => history.push('/refreshtokens')} data-testid="refreshTokens">Refresh Tokens</button>
                    </div>
                </div>
            }
            <div className="box">
                <h6 className="subtitle is-6">Lists</h6>
                <div className="buttons">
                    <button className="button is-small" onClick={() => history.push('/lists')} data-testid="lists">Lists</button>
                </div>
            </div>
        </div>
    )
}
