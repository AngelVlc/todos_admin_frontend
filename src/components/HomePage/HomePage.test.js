import { render, cleanup, fireEvent, wait } from '@testing-library/react'
import { Router } from 'react-router-dom'
import { HomePage } from './HomePage'
import { createMemoryHistory } from 'history'

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush
    })
}));

const renderWithRouter = (component) => {
    const history = createMemoryHistory();
    return {
        ...render(
            <Router history={history}>
                {component}
            </Router>)
    }
};

afterEach(cleanup);

it('should match the snapshot when the user is not logged in', () => {
    const { asFragment } = renderWithRouter(<HomePage />);

    expect(asFragment(<HomePage />)).toMatchSnapshot();
})

it('should go to users', async () => {
    const { getByTestId } = renderWithRouter(<HomePage />);

    await wait(() => {
        fireEvent.click(getByTestId('users'));
    })

    expect(mockHistoryPush.mock.calls.length).toBe(1);
    expect(mockHistoryPush.mock.calls[0][0]).toBe('/users');
})

it('should go to lists', async () => {
    const { getByTestId } = renderWithRouter(<HomePage />);

    await wait(() => {
        fireEvent.click(getByTestId('lists'));
    })

    expect(mockHistoryPush.mock.calls.length).toBe(1);
    expect(mockHistoryPush.mock.calls[0][0]).toBe('/lists');
})
