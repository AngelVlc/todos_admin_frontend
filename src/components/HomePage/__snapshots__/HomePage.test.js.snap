// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h3
      class="title"
    >
      HOME
    </h3>
    <button
      class="button"
      data-testid="lists"
    >
      Lists
    </button>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot when the user is not an admin 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h3
      class="title"
    >
      HOME
    </h3>
    <button
      class="button"
      data-testid="lists"
    >
      Lists
    </button>
  </div>
</DocumentFragment>
`;
